topaz/objects/regexpobject.py

```diff
@@ -160,7 +160,9 @@ def get_match_result(self, space, ctx, target, found):
 
     @classdef.singleton_method("allocate")
     def method_allocate(self, space):
-        return W_RegexpObject(space, None, 0)
+        w_obj = W_RegexpObject(space, None, 0)
+        w_obj.source = None
```

Comment:
Should this just be in `__init__`? Before calling `set_source` just set `self.source = None`?